src/nvim/msgpack_rpc/channel.c

```diff
@@ -313,27 +312,39 @@ static void handle_request(Channel *channel, msgpack_object *request)
     return;
   }
 
-  // Retrieve the request handler
   MsgpackRpcRequestHandler handler;
-  Array args = ARRAY_DICT_INIT;
   msgpack_object *method = msgpack_rpc_method(request);
+  handler = msgpack_rpc_get_handler_for(method->via.bin.ptr,
+                                        method->via.bin.size);
 
-  if (method) {
-    handler = msgpack_rpc_get_handler_for(method->via.bin.ptr,
-                                          method->via.bin.size);
-    if (handler.fn == msgpack_rpc_handle_missing_method) {
-      String m = method->via.bin.size > 0
-        ? cbuf_to_string(method->via.bin.ptr, method->via.bin.size)
-        : cstr_to_string("<empty>");
-      ADD(args, STRING_OBJ(m));
-      handler.async = true;
-    } else if (!msgpack_rpc_to_array(msgpack_rpc_args(request), &args)) {
-      handler.fn = msgpack_rpc_handle_invalid_arguments;
-      handler.async = true;
-    }
-  } else {
-    handler.fn = msgpack_rpc_handle_missing_method;
-    handler.async = true;
+  // check method arguments
+  Array args = ARRAY_DICT_INIT;
+  if (!msgpack_rpc_to_array(msgpack_rpc_args(request), &args)
+      && !ERROR_SET(&error)) {
```

Comment:
The first block (whichever it is) does not need the `!ERROR_SET(&error)` check, because it is always true.

Reply:
You are right, I was being conservative there! Ill replace the check with a comment that the previous block returns on Error.

Reply:
We should switch the order of the conditions here.

Reply:
> We should switch the order of the conditions here.

Switching the order in this `if`:
If `msgpack_rpc_get_handler()` has set an error, we don't need to call `msgpack_rpc_to_array(...)`. We just need to check for an error first. Less work.

Reply:
Ah, i see! I will do that.